tests/end_to_end/branch-likely-one-instruction/manual-out.c

```diff
@@ -3,8 +3,8 @@ s32 test(); // static
 s32 test(void) {
     s32 phi_a0;
 
-    phi_a0 = *(void *)0x8009A600 == 3;
-    if (*(void *)0x800B0F15 == 5) {
+    phi_a0 = *(s32 *)0x8009A600 == 3;
```

Comment:
(huh, how does it know this is a s32 and not a u32?)

Reply:
My understanding of [the code](https://github.com/matt-kempster/mips_to_c/blob/master/src/types.py#L312) is if it's either an `s32` or a `u32`, it calls it an `s32`. The `?32`s come when it's not sure if it's a `f32`, but here there's the `== 3`?

Reply:
Ah, right! And 2e08819b4ee6607be377845dca11d73fed1cdac1 (Allow `lw` to be used for any 32-bit type, not just s32) made store type annotations more `?`y, but loads haven't yet gone through that treatment